swift/Samples/TabBarContainer/DemoApp/Bar/BarBadgeWorkflow.swift

```diff
@@ -0,0 +1,120 @@
+/*
+* Copyright 2019 Square Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import Workflow
+import WorkflowUI
+import ReactiveSwift
+import TabBarContainer
+
+// MARK: Input and Output
+
+struct BarBadgeWorkflow: Workflow {
+
+    enum Output {
+        case barTabSelected
+    }
+}
+
+
+// MARK: State and Initialization
+
+extension BarBadgeWorkflow {
+
+    struct State {
+        var badgeValue: String?
+    }
+
+    func makeInitialState() -> BarBadgeWorkflow.State {
+        return State(badgeValue: nil)
+    }
+
+    func workflowDidChange(from previousWorkflow: BarBadgeWorkflow, state: inout State) {
+
+    }
+}
+
+
+// MARK: Actions
+
+extension BarBadgeWorkflow {
+
+    enum Action: WorkflowAction {
+
+        typealias WorkflowType = BarBadgeWorkflow
+        
+        case badgeValueUpdated(String?)
+        case barTabSelected
+
+        func apply(toState state: inout BarBadgeWorkflow.State) -> BarBadgeWorkflow.Output? {
+
+            switch self {
+            case .badgeValueUpdated(let badgeValue):
+                state.badgeValue = badgeValue
+                return nil
+            case .barTabSelected:
+                return .barTabSelected
+            }
+
+        }
+    }
+}
+
+
+// MARK: Workers
+
+extension BarBadgeWorkflow {
+
+    struct BarBadgeWorker: Worker {
+
+        enum Output {
+            case badgeValue(String?)
+        }
+
+        func run() -> SignalProducer<Output, Never> {
+            return SignalProducer<Output, Never>(value: .badgeValue("1"))
+        }
+
+        func isEquivalent(to otherWorker: BarBadgeWorker) -> Bool {
+            return true
+        }
+
+    }
+
+}
+
+// MARK: Rendering
+
+extension BarBadgeWorkflow {
+    typealias Rendering = BarItem
```

Comment:
If I were writing something that had a workflow for its badge value, I’d probably render out just the badge value itself rather than a whole `BarItem`. As it, this ties this workflow to how its consumer is displaying it (e.g. if later we wanted to move this to a sidebar style nav, I’d have to go change this “feature code”)

Reply:
Good suggestion, I wanted to get this type of feedback so I thought I'd throw an option out there.

Reply:
Do you like to see a separate Workflow for the BadgeValue or would you just rather use a Worker that is launched in the DemoWorkflow? I'm leaning towards the Worker that lives in the DemoWorkflow.